src/js/tabs/configuration.js

```diff
@@ -516,47 +521,68 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
+            let digitalProtocol = false;
+            switch (protocolName) {
+            case 'DSHOT150':
+            case 'DSHOT300':
+            case 'DSHOT600':
+            case 'DSHOT1200':
+            case 'PROSHOT1000':
+                digitalProtocol = true;
+
+                break;
+            default:
+
+                break;
             }
-            self.analyticsChanges['EscProtocol'] = newValue;
 
-            //hide not used setting for DSHOT protocol
-            let digitalProtocol = (escProtocolValue >= self.DSHOT_PROTOCOL_MIN_VALUE);
+            let rpmFeaturesVisible = $("input[id='dshotBidir']").is(':checked') || $("input[name='ESC_SENSOR']").is(':checked');
+
+            $('div.minthrottle').toggle(protocolConfigured && !digitalProtocol);
+            $('div.maxthrottle').toggle(protocolConfigured && !digitalProtocol);
+            $('div.mincommand').toggle(protocolConfigured && !digitalProtocol);
+            $('div.checkboxPwm').toggle(protocolConfigured && !digitalProtocol);
+            $('div.unsyncedpwmfreq').toggle(protocolConfigured && !digitalProtocol);
 
-            $('div.minthrottle').toggle(!digitalProtocol);
-            $('div.maxthrottle').toggle(!digitalProtocol);
-            $('div.mincommand').toggle(!digitalProtocol);
-            $('div.checkboxPwm').toggle(!digitalProtocol);
-            $('div.unsyncedpwmfreq').toggle(!digitalProtocol);
+            $('div.digitalIdlePercent').toggle(protocolConfigured && digitalProtocol);
+            $('.escSensor').toggle(protocolConfigured && digitalProtocol);
 
-            $('div.digitalIdlePercent').toggle(digitalProtocol);
-            $('.escSensor').toggle(digitalProtocol);
+            $('div.checkboxDshotBidir').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0") && digitalProtocol);
+            $('div.motorPoles').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0"));
+            $('div.motorPoles').toggle(protocolConfigured && rpmFeaturesVisible);
 
-            $('div.checkboxDshotBidir').toggle(semver.gte(CONFIG.apiVersion, "1.42.0") && digitalProtocol);
-            $('div.motorPoles').toggle(semver.gte(CONFIG.apiVersion, "1.42.0"));
-            //trigger change dshotBidir and ESC_SENSOR to show/hide Motor Poles tab
-            $("input[id='dshotBidir']").change(hideRpmFeatures).change();
-            $("input[name='ESC_SENSOR']").change(hideRpmFeatures);
+            $('.escMotorStop').toggle(protocolConfigured);
+
+            $('#escProtocolDisabled').toggle(!protocolConfigured);
 
             //trigger change unsyncedPWMSwitch to show/hide Motor PWM freq input
             $("input[id='unsyncedPWMSwitch']").change();
+        }
 
+        esc_protocol_e.val(PID_ADVANCED_CONFIG.fast_pwm_protocol + 1);
+        esc_protocol_e.change(function () {
+            const escProtocolValue = parseInt($(this).val()) - 1;
+
+            let newValue = undefined;
```

Comment:
Is this assignation needed? By default any new variable contains undefined, and the recommendation if you want to explicit assign a no value to a variable is to assign null.
Does the analytics require undefined and not null?

Reply:
It's not strictly needed, but in cases like these where `unassigned` is explicitly used to unassign a prior assignment (in `self.analyticsChanges['EscProtocol'] = newValue`) I find making it explicit makes it more obvious that this is intended behaviour.